include/seqan/bam_io/bam_index_bai.h

```diff
@@ -154,6 +154,151 @@ class BamIndex<Bai>
 // Functions
 // ============================================================================
 
+// ----------------------------------------------------------------------------
+// Function viewRecords()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn BamFileIn#viewRecords
+ * @brief Extract all reads overlapping a region from a BamFileIn using an index.
+ *
+ * You provide a region <tt>[regionStart, regionEnd]</tt> on the reference <tt>refID</tt>
```

Comment:
For me the function seems to extract from the region `(regionStart, regionEnd]` but samtools gives me reads from the region `[regionStart, regionEnd]`. For example, with a read of just 2 bp at position 200 (1-based) then both viewRecords and samtools view find the when `regionStart = 200; regionEnd = 200` but only samtools finds the read when  `regionStart = 201; regionEnd = 201`. So there is probably a very minor off-by-one error somewhere...

Reply:
Darn. After my benchmarks I thought samtools extraxts reads from `(reagionStart, regionEnd)` but maybe I got confused by the BED 0-based 1-based conversion. Does samtools have any documentation on this? I couldn't find a statement.

Reply:
I couldn't find it clearly specified either but in my tests samtools view always gave me an inclusive region. I tried both using (1-based) `samtools view test.bam RNAME:BEGIN-END` and a (0-based) BED file with the `-L <BED>` option.

P.S. The `-L <BED>` option in samtools does not use the BAI index, but instead reads the entire BAM... :/

Reply:
Ops.. I tricked myself. :roll_eyes: 
I read in the BED file but didn't convert to 1-based coordinates again before I used `viewRecords`.
Thanks for the benchmarking!